File: Regression/Simple Linear Regression/my_simple_linear_regression.R

# Simple Linear Regression

# Importing Dataset
dataset = read.csv('Salary_Data.csv')

# Splitting The DataSet into Training Set and Test Set
library(caTools)
set.seed(123)
split = sample.split(dataset$YearsExperience, SplitRatio = 2/3)
train_set = subset(dataset , split == TRUE)
test_set  = subset(dataset , split == FALSE)

# Feature Scaling
# train_set[, 2:3] = scale(train_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

# Fitting the Simple Linear Regression to the training set
regressor = lm(formula = Salary ~ YearsExperience,
               data = train_set)

# Predicting the Test Set Results
y_pred = predict(regressor, newdata = test_set)

# Visualising the Training Set Results
library(ggplot2)
ggplot() +
  geom_point(aes( x = train_set$YearsExperience, y = train_set$Salary),
             color = 'red') + 
  geom_line(aes( x = train_set$YearsExperience, y = predict(regressor, new_data = train_set)),
            color = 'blue') + 
  ggtitle('Salary vs Experience (Training Set)') + 
  xlab('Years Of Experience') + 
  ylab('Salary')

# Visualising the Test Set Results
library(ggplot2)
ggplot() +
  geom_point(aes( x = test_set$YearsExperience, y = test_set$Salary),
             color = 'red') + 
  geom_line(aes( x = train_set$YearsExperience, y = predict(regressor, new_data = train_set)),
            color = 'blue') + 
  ggtitle('Salary vs Experience (Test Set)') + 
  xlab('Years Of Experience') + 
  ylab('Salary')

File: Regression/Simple Linear Regression/my_simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Mon Aug 13 02:00:10 2018

@author: ritwik
"""

# Importing Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


# Importing Dataset
dataset=pd.read_csv('Salary_Data.csv')
X=dataset.iloc[:,:-1].values
Y=dataset.iloc[:,1].values


# Splitting The DataSet into Training Set and Test Set
from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=1/3, random_state=0)


# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""

# Fitting Simple Linear Regression Into Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, Y_train)

# Predicting The Test Data
y_pred = regressor.predict(X_test)

# Visualising The Training Set Results
plt.scatter(X_train, Y_train, color='red')
plt.plot(X_train,regressor.predict(X_train), color='blue')
plt.title('Salary vs Experience (Training Set)')
plt.xlabel('Years Of Experience')
plt.ylabel('Salary')
plt.show()

# Visualising The Test Set Results
plt.scatter(X_test, Y_test, color='red')
plt.plot(X_train,regressor.predict(X_train), color='blue')
plt.title('Salary vs Experience (Test Set)')
plt.xlabel('Years Of Experience')
plt.ylabel('Salary')
plt.show()
